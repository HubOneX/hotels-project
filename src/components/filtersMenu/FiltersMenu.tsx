import styled from "@emotion/styled";
import { Box, Container, Rating } from "@mui/material";
import React, { FC } from "react";
import colors from "../../constants/colors";
import { GuestType } from "../../enums/enums";
import GuestsSelection from "./guestsSelection/GuestsSelection";

type Props = {
  rating: number;
  setRating: React.Dispatch<React.SetStateAction<number>>;
  adultsCount: number;
  setAdultsCount: React.Dispatch<React.SetStateAction<number>>;
  childrenCount: number;
  setChildrenCount: React.Dispatch<React.SetStateAction<number>>;
};

const FiltersCard = styled(Box)({
  display: "flex",
  justifyContent: "center",
  alignItems: "center",
  backgroundColor: colors.lightGrey,
  borderRadius: "8px",
  boxShadow: `0px 8px 12px -8px ${colors.midGrey}`,
});

const FiltersMenu: FC<Props> = ({
  rating,
  setRating,
  adultsCount,
  setAdultsCount,
  childrenCount,
  setChildrenCount,
}) => {
  return (
    <Container
      sx={{
        display: "flex",
        justifyContent: "center",
        transform: { xs: "translateY(-85px)", sm: "translateY(-30px)" },
      }}
    >
      <FiltersCard
        sx={{
          flexDirection: { xs: "column", sm: "row" },
          width: { xs: "220px", sm: "600px" },
          padding: { xs: "20px", sm: "10px" },
          gap: { xs: "5px", sm: "22px" },
        }}
      >
        <Rating
          value={rating}
          onChange={(_, newValue) => {
            if (!newValue) return;
            setRating(newValue);
          }}
          data-testid="StarRatingFilter"
          sx={{ order: 2, marginTop: { xs: "5px", sm: 0 } }}
        />
        <GuestsSelection
          guestType={GuestType.Adult}
          currentCount={adultsCount}
          setCurrentCount={setAdultsCount}
        />
        <GuestsSelection
          guestType={GuestType.Child}
          currentCount={childrenCount}
          setCurrentCount={setChildrenCount}
          isAdditionDisabled={adultsCount === 0}
        />
      </FiltersCard>
    </Container>
  );
};

export default FiltersMenu;
