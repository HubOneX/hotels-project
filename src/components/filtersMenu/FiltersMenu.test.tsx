import { fireEvent, render, screen } from "@testing-library/react";
import FiltersMenu from "./FiltersMenu";

describe("FiltersMenu", () => {
  it("should render correctly", () => {
    render(
      <FiltersMenu
        rating={0}
        setRating={jest.fn()}
        adultsCount={0}
        setAdultsCount={jest.fn()}
        childrenCount={0}
        setChildrenCount={jest.fn()}
      />
    );

    const foundAdultFilter = screen.getByText("Adults:");
    const foundChildrenFilter = screen.getByText("Children:");
    const foundStarRating = screen.getByTestId("StarRatingFilter");

    expect(foundAdultFilter).toBeInTheDocument();
    expect(foundChildrenFilter).toBeInTheDocument();
    expect(foundStarRating).toBeInTheDocument();
  });

  it("should change star rating value on unchecked star click", () => {
    const setRatingMock = jest.fn();

    render(
      <FiltersMenu
        rating={0}
        setRating={setRatingMock}
        adultsCount={0}
        setAdultsCount={jest.fn()}
        childrenCount={0}
        setChildrenCount={jest.fn()}
      />
    );

    //MUI test id for unchecked star
    const starRating = screen.getAllByTestId("StarBorderIcon")[3];

    fireEvent.click(starRating);

    expect(setRatingMock).toBeCalledWith(4);
  });

  it("should not change star rating value on change", () => {
    const setRatingMock = jest.fn();

    render(
      <FiltersMenu
        rating={3}
        setRating={setRatingMock}
        adultsCount={0}
        setAdultsCount={jest.fn()}
        childrenCount={0}
        setChildrenCount={jest.fn()}
      />
    );

    //MUI test id for checked star
    const starRating = screen.getAllByTestId("StarIcon")[2];

    fireEvent.click(starRating);

    expect(setRatingMock).not.toBeCalled();
  });
});
