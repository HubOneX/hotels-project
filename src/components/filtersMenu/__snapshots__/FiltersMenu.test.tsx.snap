// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersMenu should match the snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-11pq5kd-MuiContainer-root"
>
  <div
    className="MuiBox-root css-rac91x"
  >
    <span
      aria-label={null}
      className="MuiRating-root MuiRating-sizeMedium css-188rs98-MuiRating-root"
      data-testid="StarRatingFilter"
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      role={null}
    >
      <label
        className="css-dqr9h-MuiRating-label"
        htmlFor=":r1:"
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarBorderIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyHidden"
        >
          1 Star
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyHidden"
        disabled={false}
        id=":r1:"
        name=":r0:"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={1}
      />
      <label
        className="css-dqr9h-MuiRating-label"
        htmlFor=":r2:"
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarBorderIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyHidden"
        >
          2 Stars
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyHidden"
        disabled={false}
        id=":r2:"
        name=":r0:"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={2}
      />
      <label
        className="css-dqr9h-MuiRating-label"
        htmlFor=":r3:"
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarBorderIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyHidden"
        >
          3 Stars
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyHidden"
        disabled={false}
        id=":r3:"
        name=":r0:"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={3}
      />
      <label
        className="css-dqr9h-MuiRating-label"
        htmlFor=":r4:"
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarBorderIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyHidden"
        >
          4 Stars
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyHidden"
        disabled={false}
        id=":r4:"
        name=":r0:"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={4}
      />
      <label
        className="css-dqr9h-MuiRating-label"
        htmlFor=":r5:"
      >
        <span
          className="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="StarBorderIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            />
          </svg>
        </span>
        <span
          className="MuiRating-visuallyHidden"
        >
          5 Stars
        </span>
      </label>
      <input
        checked={false}
        className="MuiRating-visuallyHidden"
        disabled={false}
        id=":r5:"
        name=":r0:"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={5}
      />
      <label
        className="MuiRating-label MuiRating-pristine css-dqr9h-MuiRating-label"
      >
        <input
          checked={false}
          className="MuiRating-visuallyHidden"
          id=":r0:-empty"
          name=":r0:"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value=""
        />
        <span
          className="MuiRating-visuallyHidden"
        >
          Empty
        </span>
      </label>
    </span>
    <div
      className="MuiBox-root css-1cfadgt"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        Adults
        :
      </p>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </button>
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        0
      </p>
      <button
        className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RemoveIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13H5v-2h14v2z"
          />
        </svg>
      </button>
    </div>
    <div
      className="MuiBox-root css-1cfadgt"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        Children
        :
      </p>
      <button
        className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </button>
      <p
        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        0
      </p>
      <button
        className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RemoveIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13H5v-2h14v2z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
